script.sh

```diff
@@ -2,48 +2,93 @@
+  REVIEWDOG_PATH="${TEMP_PATH}/reviewdog"
+  TFLINT_PATH="${TEMP_PATH}/tflint"
+
+  if [[ -z "${TFLINT_PLUGIN_DIR}" ]]; then
+    export TFLINT_PLUGIN_DIR="${TFLINT_PATH}/.tflint.d/plugins"
+    mkdir -p "${TFLINT_PLUGIN_DIR}"
+    echo "TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR}" >> "$GITHUB_ENV"
+  else
+    echo "Found pre-configured TFLINT_PLUGIN_DIR=${TFLINT_PLUGIN_DIR}"
+  fi
+echo '::endgroup::'
 
 echo '::group::🐶 Installing reviewdog ... https://github.com/reviewdog/reviewdog'
-curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${TEMP_PATH}" "${REVIEWDOG_VERSION}" 2>&1
+  curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b "${REVIEWDOG_PATH}" "${REVIEWDOG_VERSION}" 2>&1
+
+  echo "$REVIEWDOG_PATH" >> "$GITHUB_PATH"
+  export PATH="$REVIEWDOG_PATH:$PATH"
 echo '::endgroup::'
 
 echo '::group:: Installing tflint ... https://github.com/terraform-linters/tflint'
-curl -sfL https://raw.githubusercontent.com/terraform-linters/tflint/master/install_linux.sh | TFLINT_VERSION="${INPUT_TFLINT_VERSION}" bash
+  if [[ -z "${INPUT_TFLINT_VERSION}" ]] || [[ "${INPUT_TFLINT_VERSION}" == "latest" ]]; then
+    echo "Looking up the latest version ..."
+    version=$(curl --silent "https://api.github.com/repos/terraform-linters/tflint/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+  else
+    version=${INPUT_TFLINT_VERSION}
+  fi
+
+  curl --silent --show-error --fail \
+    --location "https://github.com/terraform-linters/tflint/releases/download/${version}/tflint_${os}_${arch}.zip" \
+    --output "${TEMP_PATH}/tflint.zip"
+
+  unzip -u "${TEMP_PATH}/tflint.zip" -d "${TEMP_PATH}/temp-tflint"
+  test ! -d "${TFLINT_PATH}" && install -d "${TFLINT_PATH}"
+  install "${TEMP_PATH}/temp-tflint/tflint" "${TFLINT_PATH}"
+  rm -rf "${TEMP_PATH}/tflint.zip" "${TEMP_PATH}/temp-tflint"
+
+  echo "${TFLINT_PATH}" >> "$GITHUB_PATH"
+  export PATH="${TFLINT_PATH}:$PATH"
 echo '::endgroup::'
 
-if [[ -z "${TFLINT_PLUGIN_DIR}" ]]; then
-  export TFLINT_PLUGIN_DIR="${TEMP_PATH}/.tflint.d/plugins"
-  echo "TFLINT_PLUGIN_DIR=$TFLINT_PLUGIN_DIR" >> "$GITHUB_ENV"
-else
-  echo "Found pre-configured TFLINT_PLUGIN_DIR=$TFLINT_PLUGIN_DIR"
-fi
 
 for RULESET in ${INPUT_TFLINT_RULESETS}; do
   PLUGIN="tflint-ruleset-${RULESET}"
   REPOSITORY="https://github.com/terraform-linters/${PLUGIN}"
 
   echo "::group:: Installing tflint plugin for ${RULESET} ... ${REPOSITORY}"
-  mkdir -p "${TFLINT_PLUGIN_DIR}"
-  curl -sfL "${REPOSITORY}"/releases/latest/download/"${PLUGIN}"_"$(uname | sed 's/Darwin/darwin/; s/Linux/linux/')"_"$(uname -m | sed 's/x86_/amd/; s/aarch64/arm/')".zip -o "${PLUGIN}".zip && unzip "${PLUGIN}".zip -d "${TFLINT_PLUGIN_DIR}" && rm "${PLUGIN}".zip
+    curl --silent --show-error --fail \
+      --location "${REPOSITORY}"/releases/latest/download/"${PLUGIN}"_"${os}"_"${arch}".zip \
```

Comment:
It looks much nicer this way 😃

Reply:
You have no idea how many iterations I went through 😆